pkg/component/component_full_description.go

```diff
@@ -0,0 +1,269 @@
+package component
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/occlient"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/storage"
+	urlpkg "github.com/openshift/odo/pkg/url"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//ComponentFullDescriptionSpec repersents complete desciption of the component
+type ComponentFullDescriptionSpec struct {
+	App        string              `json:"app,omitempty"`
+	Type       string              `json:"type,omitempty"`
+	Source     string              `json:"source,omitempty"`
+	SourceType string              `json:"sourceType,omitempty"`
+	URL        urlpkg.URLList      `json:"urls,omitempty"`
+	Storage    storage.StorageList `json:"storages,omitempty"`
+	Env        []corev1.EnvVar     `json:"env,omitempty"`
+	Ports      []string            `json:"ports,omitempty"`
+}
+
+//ComponentFullDescription describes a component fully
+type ComponentFullDescription struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ComponentFullDescriptionSpec `json:"spec,omitempty"`
+	Status            ComponentStatus              `json:"status,omitempty"`
+}
+
+func (cfd *ComponentFullDescription) copyFromComponentDesc(component *Component) error {
+	d, err := json.Marshal(component)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(d, cfd)
+}
+
+func (cfd *ComponentFullDescription) loadURLSFromClientAndLocalConfig(client *occlient.Client, localConfigInfo *config.LocalConfigInfo, componentName string, applicationName string) error {
+	urls, err := urlpkg.List(client, localConfigInfo, componentName, applicationName)
+	if err != nil {
+		return err
+	}
+
+	cfd.Spec.URL = urls
+	return nil
+}
+
+func (cfd *ComponentFullDescription) loadStoragesFromClientAndLocalConfig(client *occlient.Client, localConfigInfo *config.LocalConfigInfo, componentName string, applicationName string, componentDesc *Component) error {
```

Comment:
I feel as though this function should have more comments describing how stuff works similar to your `NewComponentFullDescriptionFromClientAndLocalConfig` function.

Reply:
added